lib/middleware/websocket/interactors/handle_message.rb

```diff
@@ -1,18 +1,31 @@
 require './lib/middleware/websocket/interactors/updates/race_update'
 require './lib/middleware/websocket/interactors/updates/timer_update'
+require './lib/middleware/websocket/interactors/handle_new_connection'
 require './lib/typinggame_server/interactors/players_rooms/update_player_room'
+require './lib/typinggame_server/interactors/players/fetch_player'
 
 module Websocket
   module Interactor
     class HandleMessage
       def call(data:, connection:)
         room_id = connection.env['PATH_INFO'][1..].to_i
+        uuid = data['uuid']
 
-        if data.key?('position')
-          update_player_position(data: data, room_id: room_id)
-          RaceUpdate.new.call(connection: connection, room_id: room_id)
-        elsif data.key?('countdown')
-          TimerUpdate.new.call(connection: connection, room_id: room_id)
+        if uuid && data.length == 1
+          Interactor::HandleNewConnection.new.call(
+            uuid: uuid, connection: connection
+          )
+        end
+
+        if data.key?('position') && data.key?('uuid')
```

Comment:
We could possibly use an entity here as `data` (instead of a Hash) so that we don't need these `if` statements. If we use specific entities we would know exactly which attributes they contain.